agent/hcp/telemetry/otlp_transform_test.go

```diff
@@ -0,0 +1,258 @@
+				IsMonotonic: false,
+				DataPoints:  otelDP,
+			},
+		},
+		{
+			Name:        "invalid-histogram",
+			Description: "Invalid histogram",
+			Unit:        "1",
+			Data: metricdata.Histogram{
+				Temporality: metricdata.DeltaTemporality,
+				DataPoints:  otelHDP,
+			},
+		},
+		{
+			Name:        "unknown",
+			Description: "Unknown aggregation",
+			Unit:        "1",
+			Data:        metricdata.Sum[int64]{},
+		},
+	}
+
+	otlpMetrics = []*mpb.Metric{
+		{
+			Name:        "float64-gauge",
+			Description: "Gauge with float64 values",
+			Unit:        "1",
+			Data:        &mpb.Metric_Gauge{Gauge: &mpb.Gauge{DataPoints: otlpDP}},
+		},
+		{
+			Name:        "float64-sum",
+			Description: "Sum with float64 values",
+			Unit:        "1",
+			Data: &mpb.Metric_Sum{Sum: &mpb.Sum{
+				AggregationTemporality: mpb.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
+				IsMonotonic:            false,
+				DataPoints:             otlpDP,
+			}},
+		},
+		{
+			Name:        "float64-histogram",
+			Description: "Histogram",
+			Unit:        "1",
+			Data: &mpb.Metric_Histogram{Histogram: &mpb.Histogram{
+				AggregationTemporality: mpb.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
+				DataPoints:             otlpHDP,
+			}},
+		},
+	}
+
+	// ScopeMetrics Test Cases
+	otelScopeMetrics = []metricdata.ScopeMetrics{{
+		Scope: instrumentation.Scope{
+			Name:    "test/code/path",
+			Version: "v0.1.0",
+		},
+		Metrics: otelMetrics,
+	}}
+
+	otlpScopeMetrics = []*mpb.ScopeMetrics{{
+		Scope: &cpb.InstrumentationScope{
+			Name:    "test/code/path",
+			Version: "v0.1.0",
+		},
+		Metrics: otlpMetrics,
+	}}
+
+	// ResourceMetrics Test Cases
+	otelResourceMetrics = &metricdata.ResourceMetrics{
+		Resource: resource.NewSchemaless(
+			semconv.ServiceName("test server"),
+			semconv.ServiceVersion("v0.1.0"),
+		),
+		ScopeMetrics: otelScopeMetrics,
+	}
+
+	otlpResourceMetrics = &mpb.ResourceMetrics{
+		Resource: &rpb.Resource{
+			Attributes: []*cpb.KeyValue{
+				{
+					Key: "service.name",
+					Value: &cpb.AnyValue{
+						Value: &cpb.AnyValue_StringValue{StringValue: "test server"},
+					},
+				},
+				{
+					Key: "service.version",
+					Value: &cpb.AnyValue{
+						Value: &cpb.AnyValue_StringValue{StringValue: "v0.1.0"},
+					},
+				},
+			},
+		},
+		ScopeMetrics: otlpScopeMetrics,
+	}
+)
+
+// TestTransformOTLP runs tests from the "bottom-up" of the metricdata data types.
+func TestTransformOTLP(t *testing.T) {
+	// Histogram DataPoint Test Case (Histograms)
+	assert.Equal(t, otlpHDP, histogramDataPoints(otelHDP))
+
+	// Number DataPoint Test Case (Counters / Gauges)
+	require.Equal(t, otlpDP, dataPoints(otelDP))
+
+	// Metrics Test Case
+	m, err := metrics(otelMetrics)
+	require.Equal(t, otlpMetrics, m)
+	require.Equal(t, len(otlpMetrics), 3)
+	require.Contains(t, err.Error(), "3 errors occurred")
+
+	// Scope Metrics Test Case
+	sm, err := scopeMetrics(otelScopeMetrics)
```

Comment:
If we can name these functions different from the struct name it might help to be a bit more descriptive as to what they do. ex: `scopeMetricsToPB`

Reply:
yep done!